geopandas/io/file.py

```diff
@@ -62,6 +63,8 @@ def read_file(filename, **kwargs):
     with reader(path_or_bytes, **kwargs) as f:
         crs = f.crs
         if bbox is not None:
+            if isinstance(bbox, GeoDataFrame):
+                bbox = tuple(bbox.to_crs(crs).total_bounds)
```

Comment:
A potential drawback here is that this will throw a surprising error if one were to pass in a frame without a defined CRS.

Reply:
Also could cause issues if file being read (somehow) didn't have a CRS defined. How would you like to handle this? Could raise a specific exception (ValueError?), but I'm tempted to just let this fail.

Reply:
I'd be inclined towards something like catching the error that is raised, and then re-raising with an error that suggests opening the file without the `bbox` argument or something of that sort. I think the present error could be hard to connect to using the `bbox` argument, since it isn't necessarily clear without looking at the code why that would cause a CRS related issue.